cmake_modules/IcebergThirdpartyToolchain.cmake

```diff
@@ -285,3 +287,20 @@ function(resolve_nlohmann_json_dependency)
 endfunction()
 
 resolve_nlohmann_json_dependency()
+
+# ----------------------------------------------------------------------
+# zlib
+
+function(resolve_zlib_dependency)
+  # use system zlib, zlib is required by arrow and avro
+  find_package(ZLIB REQUIRED)
+  if(ZLIB_FOUND)
+    list(APPEND ICEBERG_SYSTEM_DEPENDENCIES ZLIB)
+    message(STATUS "ZLIB_FOUND")
+    message(STATUS "ZLIB_LIBRARIES: ${ZLIB_LIBRARIES}")
+    message(STATUS "ZLIB_INCLUDE_DIR: ${ZLIB_INCLUDE_DIR}")
+  endif()
+
+endfunction()
+
+resolve_zlib_dependency()
```

Comment:
We need to move all of these to the top. `resolve_zlib_dependency()` should be called before `resolve_arrow_dependency()` and `resolve_avro_dependency()`